addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/MiHomeSubdevicesHandler.java

```diff
@@ -0,0 +1,648 @@
+        Configuration configuration = editConfiguration();
+        this.mihomeID = deviceID;
+        // We persist the device ID, it indicates that the device is already added in the Mi|Home cloud
+        configuration.put(MiHomeBindingConstants.PROPERTY_DEVICE_ID, this.mihomeID);
+        updateConfiguration(configuration);
+
+        updateThingProperties(newDevice);
+
+        // Update the label in the MiHome portal
+        String label = getThing().getLabel();
+        JsonObject updateResponse = gatewayHandler.updateSubdevice(this.mihomeID, label);
+        if (updateResponse == null) {
+            logger.warn("Failed to update label of newly paired subdevice {} to {}", this.mihomeID, label);
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        scheduleRegularUpdate(updateInterval);
+    }
+
+    private void scheduleRegularUpdate(long interval) {
+        if (updateTask != null) {
+            this.updateTask.cancel(true);
+            this.updateTask = null;
+        }
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    if (getThing().getStatus() == ThingStatus.ONLINE) {
+                        JsonObject data = gatewayHandler.getSubdeviceData(mihomeID);
+                        if (data != null) {
+                            // Update the channels
+                            List<Channel> channels = getThing().getChannels();
+                            for (Channel channel : channels) {
+                                ChannelUID uid = channel.getUID();
+                                if (isLinked(uid.getId())) {
+                                    updateThingState(data, uid);
+                                }
+                            }
+
+                            // Update the label if changed
+                            JsonElement newLabelElement = data.get(DeviceConstants.DEVICE_LABEL_KEY);
+                            if (newLabelElement != null) {
+                                String newLabel = newLabelElement.getAsString();
+                                String currentLabel = getThing().getLabel();
+                                if (newLabel != currentLabel) {
+                                    getThing().setLabel(newLabel);
+                                }
+                            }
+                        } else {
+                            logger.info(
+                                    "Regular update will be stopped! Device may be deleted from the MiHome Server.");
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Subdevice " + mihomeID + " is missing. It might be deleted!");
+                        }
+
+                    }
+                } catch (Exception e) {
+                    logger.error("Exception occurred during execution of update task for device with ID {} : ",
```

Comment:
I agree, I will change the status to OFFLINE, but why not logging at error level ? If exception occurred, the state of the device won't be update anymore. Is that not this case: 
>error logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action.